johnzon-mapper/src/main/java/org/apache/johnzon/mapper/MappingParserImpl.java

```diff
@@ -373,11 +377,20 @@ private Object buildObject(final Type inType, final JsonObject object, final boo
         }
 
         Object t;
-        if (classMapping.factory.getParameterTypes() == null || classMapping.factory.getParameterTypes().length == 0) {
-            t = classMapping.factory.create(null);
-        } else {
-            t = classMapping.factory.create(createParameters(classMapping, object, jsonPointer));
+        try {
+            if (classMapping.factory.getParameterTypes() == null || classMapping.factory.getParameterTypes().length == 0) {
+                t = classMapping.factory.create(null);
+            } else {
+                t = classMapping.factory.create(createParameters(classMapping, object, jsonPointer));
+            }
+        } catch (Exception e) {
+            final String message = String.format("%s cannot be constructed to deserialize %s: %s%n%s",
```

Comment:
can it be a mapper exception (normally yes from create parameter
do we want to bypass this catch adding a catch(MapperException) before?

Reply:
I'm my testing I haven't seen any scenarios where MappingException is thrown.   Issues appear to get wrapped as IllegalStateException before reaching this catch.

Reply:
Hmm, this can need to be refined if we get 2 different mapper exceptions in the stack and avoid this ISE (in particular cause we go 1.3 now).

Reply:
Pushed something that should prevent us from doing this wrapping more than once